OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -236,6 +245,28 @@ void ITick.Tick(Actor self)
 		}
 
 		protected virtual void Tick(Actor self)
+		{
+			// PERF: Avoid LINQ when checking whether all production traits are disabled/paused
+			var anyEnabledProduction = false;
+			var anyUnpausedProduction = false;
+			foreach (var p in productionTraits)
+			{
+				if (!anyEnabledProduction)
```

Comment:
These checks aren't needed, it is more efficient to use `anyEnabledProduction |= !p.IsTraitDisabled;` and so on.  Can you please change this in all the places you use this pattern?

Reply:
Apart from the two here, I'm not aware of any other cases?